intake/source/cache.py

```diff
@@ -139,13 +141,20 @@ def load(self, urlpath):
                 logger.debug("Cached at: {}".format(cache_path))
                 self._log_metadata(urlpath, file_in.path, cache_path)
 
-                with file_in as f1:
-                    with file_out as f2:
-                        data = True
-                        while data:
-                            #TODO: print out progress
-                            data = f1.read(self.blocksize)
-                            f2.write(data)
+                file_size = file_in.fs.size(file_in.path)
+                percent_per_block = 100 * self.blocksize / file_size
+                print("Caching {}".format(file_in.path))
+                with tqdm_notebook(total=100) as pbar:
+                    with file_in as f1:
+                        with file_out as f2:
+                            data = True
+                            while data:
+                                data = f1.read(self.blocksize)
+                                f2.write(data)
+                                pbar.update(int(percent_per_block))
+                clear_output()
```

Comment:
I assume this would remove everything from the output area of the cell - and it looks like this happens after each file is done.

Reply:
I should say, it *may* fail for HTTP